crates/valence_core/src/packet/decode.rs

```diff
@@ -125,23 +127,34 @@ impl PacketDecoder {
     }
 
     #[cfg(feature = "encryption")]
-    pub fn enable_encryption(&mut self, key: &[u8; 16]) {
+    pub fn enable_encryption(&mut self, key: &[u8; 16]) -> anyhow::Result<()> {
         assert!(self.cipher.is_none(), "encryption is already enabled");
 
-        let mut cipher = Cipher::new(key.into(), key.into());
+        let mut cipher = Cipher::new_from_slices(key, key)?;
 
         // Don't forget to decrypt the data we already have.
-        cipher.decrypt(&mut self.buf);
+        Self::decrypt_bytes(&mut cipher, &mut self.buf);
 
         self.cipher = Some(cipher);
+
+        Ok(())
+    }
+
+    /// Decrypts the provided byte slice in place using the cipher, without consuming the cipher.
+    #[cfg(feature = "encryption")]
+    fn decrypt_bytes(cipher: &mut Cipher, bytes: &mut [u8]) {
+        for chunk in bytes.chunks_mut(Cipher::block_size()) {
+            let gen_arr = GenericArray::from_mut_slice(chunk);
+            cipher.decrypt_block_mut(gen_arr);
+        }
```

Comment:
Might be able to express this more succinctly with `(en|de)crypt_blocks_inout_mut`, but it's probably fine either way.

Reply:
ngl I was having a hard time with the docs for these packages because they don't have many good examples. I'll see what I can do.

Reply:
I looked at it a bit more and I'm not convinced that inout doesn't do a copy, so lets leave it as is.